src/engraving/playback/playbackmodel.cpp

```diff
@@ -241,69 +241,102 @@ void PlaybackModel::updateEvents(const int tickFrom, const int tickTo, const tra
 {
     std::set<ID> changedPartIdSet = m_score->partIdsFromRange(trackFrom, trackTo);
 
-    for (const mu::engraving::RepeatSegment* repeatSegment : repeatList()) {
-        int tickPositionOffset = repeatSegment->utick - repeatSegment->tick;
+    for (const RepeatSegment* repeatSegment : repeatList()) {
+        int repeatTickOffset = repeatSegment->utick - repeatSegment->tick;
         int repeatStartTick = repeatSegment->tick;
         int repeatEndTick = repeatStartTick + repeatSegment->len();
 
         if (repeatStartTick > tickTo || repeatEndTick <= tickFrom) {
             continue;
         }
 
-        for (const mu::engraving::Measure* measure : repeatSegment->measureList()) {
-            int measureStartTick = measure->tick().ticks();
-            int measureEndTick = measure->endTick().ticks();
+        const auto& measureList = repeatSegment->measureList();
 
-            if (measureStartTick > tickTo || measureEndTick <= tickFrom) {
+        for (const Staff* staff : m_score->staves()) {
+            ID partId = staff->part()->id();
+
+            if (changedPartIdSet.find(partId) == changedPartIdSet.cend()) {
                 continue;
             }
 
-            for (mu::engraving::Segment* segment = measure->first(); segment; segment = segment->next()) {
-                if (!segment->isChordRestType()) {
-                    continue;
-                }
+            const staff_idx_t staffIdx = staff->idx();
+            const auto& staffMeasureList = repeatSegment->measureList(staffIdx);
 
-                int segmentStartTick = segment->tick().ticks();
-                int segmentEndTick = segmentStartTick + segment->ticks().ticks();
+            Q_ASSERT(staffMeasureList.size() == measureList.size());
 
-                if (segmentStartTick > tickTo || segmentEndTick <= tickFrom) {
-                    continue;
-                }
+            for (size_t i = 0; i < measureList.size(); ++i) {
+                const Measure* measure = measureList.at(i);
+                const Measure* staffMeasure = staffMeasureList.at(i);
+
+                int measureTicksOffset = repeatTickOffset + (measure->tick().ticks() - staffMeasure->tick().ticks());
```

Comment:
This is still quite ugly, but it is necessary to get `measureTicksOffset`. Has anyone a suggestion how I could solve this in a different way?